### PR TITLE
Automate downloading and installing the correct license plugin if your system is now locked on the license screen

### PR DESCRIPTION
**Pull request by mikecirioli**

### CODE REVIEW

File: license-upgrade/core-traditional-unbreak.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+# backup the currently installed plugin
+echo "Backing up the currently installed license plugin"
+mv $JENKINS_HOME/plugins/cloudbees-license.jpi $JENKINS_HOME/plugins/cloudbees-license.bak
+
+# now download the plugin
+
+echo "Downloading updated plugin from $PLUGIN_URL"
+if [ -x "$(which wget)" ] ; then
+    wget -o $JENKINS_HOME/plugins/cloudbees-license.jpi $PLUGIN_URL
```
Comment (escoem) [inline license-upgrade/core-traditional-unbreak.sh:112]: According to documentation, it is `-O` 

```suggestion
    wget -O $JENKINS_HOME/plugins/cloudbees-license.jpi $PLUGIN_URL
```

File: license-upgrade/core-traditional-unbreak.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+hput versions "9.27" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.27.1/cloudbees-license.hpi"
+hput versions "9.26" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.26.1/cloudbees-license.hpi"
+hput versions "9.24" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.24.1/cloudbees-license.hpi"
+hput versions "9.20" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.20.1/cloudbees-license.hpi"
+hput versions "9.18.1" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.18.1.1/cloudbees-license.hpi"
+#TODO: verify this is not a mistake - 9.18.1 is not a valid backport for the new licenses....
+hput versions "9.18" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.18.1/cloudbees-license.hpi"
+hput versions "9.17" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.17.1/cloudbees-license.hpi"
+#TODO: verify this is not a mistake
+hput versions "9.24" "https://jenkins-updates.cloudbees.com/download/plugins/cloudbees-license/9.24.1/cloudbees-license.hpi"
```
[the remark this answers is not in the document]
Comment (mikecirioli) [reply license-upgrade/core-traditional-unbreak.sh:33]: human error i believe, double checking against https://support.cloudbees.com/hc/en-us/articles/360042376431-Upgrading-for-the-new-CloudBees-License-Certificate#locatetherequiredversionofthecloudbeeslicensemanagerplugin now

File: license-upgrade/core-traditional-unbreak.sh [hunk 3]
```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+## Script to automatically determine what version of plugin needs to be downloaded
+## and installs it
+
```
[the remark this answers is not in the document]
Comment (mikecirioli) [reply license-upgrade/core-traditional-unbreak.sh:5]: +1

File: license-upgrade/core-traditional-unbreak.sh [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+hget() {
+    PLUGIN_URL=$(grep "^$2 " /tmp/hashmap.$1 | awk '{ print $2 };' )
+}
+
+verify_command() {
+  echo "Verifying command [${1}] is installed..."
+
+  if command --version $1 >/dev/null 2>&1 || command -version $1 >/dev/null 2>&1 || command -v $1 >/dev/null 2>&1; then
```
Comment (amuniz) [inline license-upgrade/core-traditional-unbreak.sh:21]: Wouldn't be easier to use `which`?

Comment (amuniz) [reply license-upgrade/core-traditional-unbreak.sh:21]: AFAICT it's a kernel built-in one, so granted to be available.

Comment (mikecirioli) [reply license-upgrade/core-traditional-unbreak.sh:21]: on a mac its an actual program apparently

Comment (mikecirioli) [reply license-upgrade/core-traditional-unbreak.sh:21]: so i did some reading, and for bash, it seems `command` is the preferred way to do it
https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then/85250#85250

**Comment by mikecirioli** [comment]
superseded by #156

**Pull request closed without merging**